Tests/CacheAdvanceTests/CacheAdvanceTests.swift

```diff
@@ -125,6 +125,80 @@ final class CacheAdvanceTests: XCTestCase {
         }
     }
 
+    private func prepareCacheWithIncorrectOffsetInHeader() throws -> CacheHeaderHandle {
+        let randomHighValue: UInt64 = 10_1000
+        let header = try CacheHeaderHandle(
+            forReadingFrom: testFileLocation,
+            maximumBytes: randomHighValue,
+            overwritesOldMessages: true)
+        let cacheToWrite = CacheAdvance<TestableMessage>(
+            fileURL: testFileLocation,
+            writer: try FileHandle(forWritingTo: testFileLocation),
+            reader: try CacheReader(forReadingFrom: testFileLocation),
+            header: try CacheHeaderHandle(
+                forReadingFrom: testFileLocation,
+                maximumBytes: header.maximumBytes,
+                overwritesOldMessages: header.overwritesOldMessages),
+            decoder: JSONDecoder(),
+            encoder: JSONEncoder())
+
+        let message = TestableMessage(stringLiteral: TestableMessage.lorumIpsum[0].value)
+        try cacheToWrite.append(message: message)
+
+        // Make sure the header data is persisted before we read it as part of the `messages()` call below.
+        try header.synchronizeHeaderData()
+        // Our file is empty. Make the file corrupted by setting the offset at end of newest message to be further in the file.
+        // This should never happen, but past versions of this repo could lead to a file having this kind of inconsistency if a crash occurred at the wrong time.
+        try header.updateOffsetInFileAtEndOfNewestMessage(to: header.offsetInFileAtEndOfNewestMessage + 1000)
```

Comment:
Digging in a bit myself here, I confirmed the that value of `header.offsetInFileAtEndOfNewestMessage` is 203.

<img width="2732" alt="Screen Shot 2022-10-10 at 4 08 06 PM" src="https://user-images.githubusercontent.com/1791049/194965334-e7205910-1356-48cc-a715-ae591c6dd6bd.png">


Originally when we said in the test that the "offset in file at end of newest message is larger than the actual file length" I thought that the offset was larger than the maximum bytes, i.e. larger than 101,000.

Now I see that we are saying that the offset is larger than the size of the file on disk, if I'm understanding correctly.

Reply:
>I see that we are saying that the offset is larger than the size of the file on disk
Yes, this is root cause.